From 841084a6c6c9dfc3cbc2aa9f67a0162be2667f3a Mon Sep 17 00:00:00 2001
From: Subbaraya Sundeep <sbhatta@marvell.com>
Date: Thu, 7 Feb 2019 18:54:16 +0530
Subject: [PATCH 0959/1051] octeontx2-af: Do not clean up if init is successful

Though rvu init is successful ptp and cgx drivers are
getting unregistered hence fix it by returning zero.

Change-Id: I2ac81971db7f04dc863f0712930ad6342151c87f
Signed-off-by: Subbaraya Sundeep <sbhatta@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/3653
Tested-by: sa_ip-sw-jenkins
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/rvu.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu.c
index 8f92945bcafc..e27f434a04da 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu.c
@@ -2689,6 +2689,7 @@ static int __init rvu_init_module(void)
 	if (err < 0)
 		goto rvu_err;
 
+	return 0;
 rvu_err:
 	pci_unregister_driver(&ptp_driver);
 ptp_err:
-- 
2.17.1

